spec/factories/claim/base_claims.rb

```diff
@@ -1,5 +1,34 @@
 FactoryBot.define do
 
+  trait :advocate_base_setup do
+    # NOTE: this was introduced because it was the only way to get FactoryBot to set
+    # model attributes on initialize (which seems not to be the default behaviour) and
+    # was causing the factory not to assign the appropriate attributes/associations on
+    # initialize which makes after_initialize logic not to behave as expected since the
+    # expected values are not yet set.
+    # More details can be found here:
+    # https://stackoverflow.com/questions/5916162/problem-with-factory-girl-association-and-after-initialize
+    initialize_with { new(attributes) }
+    transient do
+      create_defendant_and_rep_order true
+    end
+
+    form_id SecureRandom.uuid
+    court
+    case_number { random_case_number }
+    external_user
+    source { 'web' }
+    apply_vat false
+    providers_ref { random_providers_ref }
+    case_type
+    offence
+    advocate_category 'QC'
+    sequence(:cms_number) { |n| "CMS-#{Time.now.year}-#{rand(100..199)}-#{n}" }
+
+    after(:build) { |claim| post_build_actions_for_draft_claim(claim) }
```

Comment:
Is this also common setup for both advocate claims? I'm personally of the approach of not adding any behaviour unless we're 💯 sure we need it. I don't understand the complexity of `post_build_actions_for_draft_claim` but seems like calling that for *advocate interim claims* after build might just end up creating data that is not necessary and makes things harder to change in the future.

Reply:
will look into this...